src/coreclr/jit/fgdiagnostic.cpp

```diff
@@ -4737,412 +4677,18 @@ void Compiler::fgDebugCheckSsa()
 //
 void Compiler::fgDebugCheckLoopTable()
 {
-    if ((m_loops != nullptr) && optLoopsRequirePreHeaders)
+    if (m_loops == nullptr)
```

Comment:
Looks like the header comment above needs to be updated?

Reply:
Done. Also rephrased a few other things (removed references to "loop table") and deleted the loop flag enum that had been left over.